src/polyfills/formdata-entries.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getFormAsObject} from '../form';
+import {map} from '../utils/object';
+
+/**
+ * Creates a new `FormData` object.
+ *
+ * This patches the native constructor so that the polyfill can capture the
+ * fields in the form passed to the constructor. If `FormData#entries` needs a
+ * polyfill, chances are there are no methods to read the content of the
+ * `FormData` after construction, so the only way to partially polyfill
+ * `entries` is to patch the constructor (and the `append` method).
+ *
+ * For more details on this, see http://mdn.io/FormData.
+ *
+ * @param {!HTMLFormElement=} form An HTML `<form>` element — when specified,
```

Comment:
Prefix optional params with `opt_`, i.e. `opt_form`.

Reply:
Done. Did we intentionally deviate from the Google JS Style Guide on this one?

>  Optional parameters must include spaces on both sides of the equals operator, be named exactly like required parameters (i.e., not prefixed with opt_)

https://google.github.io/styleguide/jsguide.html#features-functions-default-parameters